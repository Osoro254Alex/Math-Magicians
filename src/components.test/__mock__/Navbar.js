import React from 'react';
import { BrowserRouter as Router, NavLink } from 'react-router-dom';
import styles from './Navbar.module.css';

class Navbar extends React.PureComponent {
  render() {
    return (
      <nav>
        <Router>
          <h1 className={styles.header}>Math Magicians</h1>
          <ul className={styles.links}>
            <li>
              <NavLink to="/">Home</NavLink>
            </li>
            |
            <li>
              <NavLink to="/Calculator">Calculator</NavLink>
            </li>
            |
            <li>
              <NavLink to="/Quote">Quote</NavLink>
            </li>
          </ul>
        </Router>
      </nav>
    );
  }
}

export default Navbar;
