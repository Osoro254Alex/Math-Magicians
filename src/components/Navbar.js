import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './Navbar.module.css';

class Navbar extends React.PureComponent {
  render() {
    return (
      <nav>
        <h1 className={styles.header}>Math Magicians</h1>
        <ul className={styles.links}>
          <li>
            <NavLink to="/">Home</NavLink>
          </li>
          |
          <li>
            <NavLink to="/Calculator">Calculator</NavLink>
          </li>
          |
          <li>
            <NavLink to="/Quote">Quote</NavLink>
          </li>
        </ul>
      </nav>
    );
  }
}

export default Navbar;
