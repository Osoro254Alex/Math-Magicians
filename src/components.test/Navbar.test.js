import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Navbar from './__mock__/Navbar';

afterEach(cleanup);

describe('render navbar component without error', () => {
  test('navbar should render correctly', () => {
    render(<Navbar />);
  });
  test('matches snapshot for new Header', () => {
    const tree = renderer.create(<Navbar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
