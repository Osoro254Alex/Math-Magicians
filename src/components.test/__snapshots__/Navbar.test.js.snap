// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render navbar component without error matches snapshot for new Header 1`] = `
<nav>
  <h1
    className="header"
  >
    Math Magicians
  </h1>
  <ul
    className="links"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    |
    <li>
      <a
        className=""
        href="/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    |
    <li>
      <a
        className=""
        href="/Quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
